_posts/2020-06-23-Durr_Hoyer_Post_1.md

```diff
@@ -0,0 +1,328 @@
+
+                          y_prime= Algorithm.simulate(N,y)
+
+                      else: 
+
+                          y_prime=Algorithm_Odd.simulate(N,y)
+
+                  if t[y_prime] < t[y]: # Check if we have found our minimum
+
+                      i=0 # reset loop if found
+
+              y = y_prime #store index with minimum
+
+          return y                                                      
+
+
+------------------
+
+### Motivation
+The motivation behind this project is to provide open source functionality. 
+The efficiency that is proposed by this algorithm is much better than a typical algorithm for finding the minimum of an unsorted table.
+On a classical computer it will take as many time steps as there are items to find a minimum. 
+This means the Big-O is N.
+The Dürr-Høyer algorithm takes this problem and solves it in sqrt(N) time steps.
+For example assume there are 9 items in a table, on a classical comptuer this would take 9 time steps.
+The Dürr-Høyer algorithm would find the minimum in 3 steps.
+We can understand this effecieny as the Big-O. 
+The Big-O essentially means the run time of the program will increase as the table size grows. 
+Dürr and Høyer propose a quantum algorithms in order to find the minimum of an unsorted table with a 50% success rate. 
+This success rate is bounded by the Big-O.
+If we decide to alleviate this Big-O the probability of finding the minimum increases. 
+Essentialy if we let the algorithm run for longer than the time it takes to guarantee a 50% success rate, we will get a success rate greater than 50 %. 
+With this comes some complexity in the inner workings of the library which will be explained when analyzing the algorithm. 
+In order for this library to be used properly it must meet some guidelines. 
+At the moment I am referencing the amazing template given by Dr. Sarah Kaiser https://github.com/crazy4pi314/qsharp-library-template in order to make this library usable in Q#. 
+I am programming the algorithm in Q# with a python host script.
+The circuit's gates will be analyzed and some code is given.  
+https://github.com/mridulsar/DurrHoyerLibrary   
+
+### My Background
+I began my journey from math to computers through my passion for ML algorithms, where I stumbled upon quantum computing. 
+I was instantly pulled in by the elegant algorithms and proposed efficiency.
+Recently, I started looking into quantum machine learning where you can model nodes of neural networks as qubits. 
+I took a deep dive into the research papers and found myself amazed by what was out there.
+I began with a Coursera course from St. Petersburg University on quantum computing with detail on quantum algorithm design and quantum computer architecture. 
+From here I tinkered with Q# and developed basic algorithms that I previously learned.
+I found Dürr and Høyer's paper and saw it as an fusion of Duestch's, Shor's, and Grover's Algorithm; a perfect next step for me.

```

Comment:
I think that this is fine either here or at the beginning, but there is a fair bit of grammatical/spelling errors. Maybe copy +paste into Microsoft word to have it highlight stuff?

Reply:
Sounds good, I will adjust this

Reply:
i felt it would be better at the end since it isn't not vital info and I did not want to deter readers from boring info not related to the algorithm